File: spring_java_orm_sample/src/main/java/com/example/demo/data/domain/PurchaseInvoiceType.java
package com.example.demo.data.domain;

import com.fasterxml.jackson.annotation.JsonCreator;
import org.seasar.doma.Domain;

@Domain(valueType = int.class, factoryMethod = "of")
public enum PurchaseInvoiceType {
    PURCHASE(1),
    RETURN_PURCHASE(2);

    private final int value;

    PurchaseInvoiceType(int value) {
        this.value = value;
    }

    public int getValue() {
        return value;
    }

    @JsonCreator
    public static PurchaseInvoiceType of(int value) {
        for (PurchaseInvoiceType enm : values()) {
            if (enm.value == value) {
                return enm;
            }
        }
        throw new IllegalArgumentException("Unknown PurchaseInvoiceType value: " + value);
    }
}

File: codex_poting/kotlin/src/main/kotlin/com/example/demo/data/domain/PurchaseInvoiceType.kt
package com.example.demo.data.domain

import com.fasterxml.jackson.annotation.JsonCreator
import org.seasar.doma.Domain

@Domain(valueType = Int::class, factoryMethod = "of")
enum class PurchaseInvoiceType(val value: Int) {
    PURCHASE(1),
    RETURN_PURCHASE(2);

    companion object {
        @JvmStatic
        @JsonCreator
        fun of(value: Int): PurchaseInvoiceType {
            for (enm in entries) {
                if (enm.value == value) {
                    return enm
                }
            }
            throw IllegalArgumentException("Unknown PurchaseInvoiceType value: " + value)
        }
    }
}
